backend/src/utils/route-security.ts

```diff
@@ -20,26 +21,51 @@ const usernameTranslate = (username: string): string => {
     .toLowerCase();
 };
 
-const getNamespaces = (fastify: KubeFastifyInstance): string => {
+const getNamespaces = (
+  fastify: KubeFastifyInstance,
+): { dashboardNamespace: string; notebookNamespace: string } => {
   const config = getDashboardConfig();
   const notebookNamespace = config.spec.notebookController?.notebookNamespace;
   const fallbackNamespace = config.metadata.namespace || fastify.kube.namespace;
 
-  return notebookNamespace || fallbackNamespace;
+  return {
+    notebookNamespace: notebookNamespace || fallbackNamespace,
+    dashboardNamespace: fallbackNamespace,
+  };
 };
 
 const requestSecurityGuard = async (
   fastify: KubeFastifyInstance,
   request: OauthFastifyRequest,
   name: string,
   namespace: string,
+  needsAdmin: boolean,
 ): Promise<void> => {
-  const notebookNamespace = getNamespaces(fastify);
+  const { notebookNamespace, dashboardNamespace } = getNamespaces(fastify);
   const username = await getUserName(fastify, request);
+  const isAdmin = await isUserAdmin(fastify, username, dashboardNamespace);
   const translatedUsername = usernameTranslate(username);
 
+  const isReadRequest = request.method.toLowerCase() === 'get';
+
+  // User is an admin, trust
+  if (needsAdmin && isAdmin) {
+    return;
+  }
```

Comment:
Shouldn't this be something like this?
```suggestion
  if (needsAdmin) {
    if (isAdmin) {   // User is an admin, trust
      return;
    }
    // User is not an admin, reject
   throw createCustomError(...)
  }
```

Reply:
It will fallback either way to the error as long as we don't whitelist any other rule that needs admin privileges.

Reply:
taking a random rule as an example,
```
  if (namespace === dashboardNamespace && name === `${notebookNamespace}-image-pullers`)) {
```

then this request would be allowed, wouldn't it?
- `needsAdmin` and
-  is `not(isAdmin)` and
- `namespace === dashboardNamespace` 
- `name === ${notebookNamespace}-image-pullers`

Basically, the security requirement says: `if the route needs admin rights and the user isn't admin, reject`
but what's currently implemented is:
- if not `(needsAdmin && isAdmin)`
- and not `(isReadRequest && namespace === dashboardNamespace)` 
- and not `(namespace === dashboardNamespace && name === `${notebookNamespace}-image-pullers`)`
- ...
- then reject

Reply:
Yeah, you're right, in that instance is safer to throw inside the function, let's see what @andrewballantyne thinks.

Reply:
Thanks guys, I have pushed an update to the code based on your conversation & a follow up one I had with Lucas.